flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/resource/UserResourceManagerTest.java

```diff
@@ -0,0 +1,186 @@
+        UserResourceManager userResourceManager = createResourceManager(new URL[0]);
+        URLClassLoader userClassLoader = userResourceManager.getUserClassLoader();
+
+        // test class loading before register resource
+        try {
+            Class.forName(LOWER_UDF_CLASS, false, userClassLoader);
+            fail("Should fail.");
+        } catch (ClassNotFoundException e) {
+        }
+
+        // register the same jar repeatedly
+        userResourceManager.registerResource(new ResourceUri(ResourceType.JAR, udfJar.getPath()));
+        userResourceManager.registerResource(new ResourceUri(ResourceType.JAR, udfJar.getPath()));
+
+        // assert resource infos
+        Map<ResourceUri, URL> expected =
+                Collections.singletonMap(
+                        new ResourceUri(ResourceType.JAR, udfJar.getPath()),
+                        userResourceManager.getURLFromPath(new Path(udfJar.getPath())));
+
+        assertEquals(expected, userResourceManager.getResources());
+
+        // test load class
+        final Class<?> clazz1 = Class.forName(LOWER_UDF_CLASS, false, userClassLoader);
+        final Class<?> clazz2 = Class.forName(LOWER_UDF_CLASS, false, userClassLoader);
+
+        assertEquals(clazz1, clazz2);
+    }
+
+    @Test
+    public void testRegisterInvalidResource() throws Exception {
+        UserResourceManager userResourceManager = createResourceManager(new URL[0]);
+
+        // test register non-exist file
+        final String fileUri = temporaryFolder.getRoot().getPath() + Path.SEPARATOR + "test-file";
+
+        CommonTestUtils.assertThrows(
+                String.format("Resource [%s] not found.", fileUri),
+                IllegalArgumentException.class,
+                () -> {
+                    userResourceManager.registerResource(
+                            new ResourceUri(ResourceType.FILE, fileUri));
+                    return null;
+                });
+
+        // test register directory for jar resource
+        final String jarUri = temporaryFolder.newFolder("test-jar-dir").getPath();
+
+        CommonTestUtils.assertThrows(
+                String.format(
+                        "Directory [%s] is not allowed for registering jar resource.", jarUri),
+                IllegalArgumentException.class,
+                () -> {
+                    userResourceManager.registerResource(new ResourceUri(ResourceType.JAR, jarUri));
+                    return null;
+                });
+    }
+
+    @Test
+    public void testDownloadResource() throws Exception {
+        Path srcPath = new Path(udfJar.getPath());
+        UserResourceManager userResourceManager = createResourceManager(new URL[0]);
+
+        // test download resource to local path
+        URL localUrl = userResourceManager.downloadResource(srcPath);
+
+        String expected = new String(FileUtils.readAllBytes(udfJar.toPath()));
+        String actual = new String(FileUtils.readAllBytes(Paths.get(localUrl.toURI())));
```

Comment:
This doesn't resolve my initial concern that the bytes may not be compliant with UTF-8 encoding. You can simply by comparing bytes using `assertArrayEquals()`. Do not need to convert into `String`.